src/associated-items.rst

```diff
@@ -48,6 +48,29 @@ An :t:`associated type` shall not be used in the :t:`path expression` of a
 An :t:`associated type` with a :s:`TypeBoundList` shall appear only as an
 :t:`associated trait type`.
 
+:dp:`fls_PeD0DzjK57be`
+A :t:`generic associated type` is an :t:`associated type` with
+:t:`[generic parameter]s`.
+
+:dp:`fls_Y7yIkAbxJpvp`
+A :t:`generic associated type` shall require a :t:`lifetime parameter` when
+the related :t:`implementation` or :t:`trait` has at least one
+:t:`associated function` that imposes a :t:`lifetime bound` on the
+:t:`generic associated type`.
+
+:dp:`fls_1DZOklVBn2AZ`
+A :t:`generic associated type` shall be subject to a :t:`where clause` of
+the form ``where Self: 'lifetime_parameter``, where ``lifetime_parameter`` is
+the :t:`lifetime parameter` of the :t:`generic associated type`, when the
+the related :t:`implementation` or :t:`trait` has at least one
+:t:`associated function` that imposes a :t:`lifetime bound` on the
+:t:`generic associated type`, and the intersection of all such imposed
+:t:`[lifetime bound]s` is not empty.
```

Comment:
This is not flexible enough, you can have multiple lifetimes that are subject to this, I cooked up an example showing this off a bit better https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=383528df6bff97380a3ac5516aa8b7ab

Basically, if a lifetime parameter of a generic associated type receives a lifetime argument in a function declared in that same trait that is used as a bound for `Self`, that is either an explicit or implicit `Self: 'lifetime` bound, then that generic associated type has to also declare that `Self: 'lifetime` bound for that lifetime parameter.

Reply:
Fixed (hopefully).